crates/sui-node/src/lib.rs

```diff
@@ -184,13 +175,28 @@ impl SuiNode {
         )
         .await;
 
-        let (checkpoint_executor_handle, reconfig_channel) = CheckpointExecutor::new(
-            state_sync_handle.subscribe_to_synced_checkpoints(),
-            checkpoint_store.clone(),
-            state.clone(),
-            &prometheus_registry,
-        )
-        .start()?;
+        let state_sync = if config.p2p_config.is_some() {
+            let (p2p_network, discovery_handle, state_sync_handle) =
+                Self::create_p2p_network(config, state_sync_store, &prometheus_registry)?;
+
+            let (checkpoint_executor_handle, reconfig_channel) = CheckpointExecutor::new(
+                state_sync_handle.subscribe_to_synced_checkpoints(),
+                checkpoint_store.clone(),
+                state.clone(),
+                &prometheus_registry,
+            )
+            .start()?;
+
+            Some(StateSync {
+                _p2p_network: p2p_network,
+                _discovery: discovery_handle,
+                state_sync: state_sync_handle,
+                _checkpoint_executor_handle: checkpoint_executor_handle,
+                reconfig_channel,
+            })
+        } else {
+            None
+        };
```

Comment:
This is going to have some not-so-great impacts on user's running fullnodes. In particular they're now going to need to change their config to explicitly enable state-sync vs explicitly opting out of it (which would be my preference)